typescript/22-quad-tree/quad-tree.ts

```diff
@@ -0,0 +1,31 @@
+import {Node} from "./trees.js"
+
+export function construct(grid: number[][]): Node {
+  function buildTree(row: number, col: number, size: number): Node {
+    const val: number = grid[row][col];
+    let isSame = true;
+    for (let i = row; i < row + size; i++) {
+      for (let j = col; j < col + size; j++) {
+        if (grid[i][j] !== val) {
```

Comment:
So no matter how big our grid is, we always scan it fully, on each level of recursion, just to make sure values aren't the same. This is in addition to recursive calls. We can do better.